go/vt/vtgate/executor.go

```diff
@@ -884,6 +886,45 @@ func (e *Executor) showVitessReplicationStatus(ctx context.Context, filter *sqlp
 	}, nil
 }
 
+func (e *Executor) getMaxReplicationLag(session *SafeSession) sqltypes.Value {
+	status := e.scatterConn.GetHealthCheckCacheStatus()
+	replLag := -1
+	targetString := session.GetTargetString()
```

Comment:
If user doesn't specify a tablet_type, does this default to `PRIMARY`?
For a default connection aka primary connection, we want lag to be reported as 0.
Let us add a test to verify this use case.

Reply:
if no tablet_type is provided then it will go through all the tablets and report maximum replication delay.
Only when a specific tablet_type is provided via targetString then only for that tablet the replication delay will be reported.

Reply:
If I connect via mysql client to vtgate and don't issue a `use`, is tablet_type set?